### ISSUE TITLE
BSON.Value rawValue initializer does not recognize arrays

### ISSUE THREAD

**Issue by svanimpe** [Danappelxx/SwiftMongoDB#20]
The following test does not succeed with a `rawValue` of `[-73, 40]`:

```
if let rawArray = rawValue as? [Any]
```

This is using the March 1 build of Swift on Ubuntu 15.10

**Comment by svanimpe** [comment]
Looks like `Any` can't be cast to `[Any]` for some reason. A more specific cast to `[Int]` does work.

**Comment by Danappelxx** [comment]
Yeah - all of this is an issue in a pure-swift environment because it doesn't have objective-c to do some higher-level type casting. I'm not sure what the Swift team is going to do about this but for now I think it's safest to remove the rawValue initializer so that people don't get surprised by this sort of behavior.
